addons/mrp_repair/models/mrp_repair.py

```diff
@@ -0,0 +1,539 @@
+        if self.product_id:
+            self.guarantee_limit = False
+            self.lot_id = False
+            self.product_uom_id = self.product_id.uom_id.id
+
+    @api.onchange('product_uom_id')
+    def onchange_product_uom_id(self):
+        res = []
+        if not self.product_uom_id or not self.product_id:
+            return res
+        if self.product_uom_id.category_id.id != self.product_id.uom_id.category_id.id:
+            raise UserError('The Product Unit of Measure you chose has a different category than in the product form.')
+            self.product_uom_id = self.product_id.uom_id.category_id.id
+
+    @api.onchange('location_id')
+    def onchange_location_id(self):
+        """ On change of location
+        """
+        self.location_dest_id = self.location_id.id
+
+    @api.multi
+    def button_dummy(self):
+        return True
+
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        """ On change of partner sets the values of partner address,
+        partner invoice address and pricelist.
+        @param part: Changed id of partner.
+        @param address_id: Address id from current record.
+        """
+        if not self.partner_id:
+            self.address_id = False
+            self.partner_invoice_id = False
+            self.pricelist_id = self.pricelist_id.search([('type', '=', 'sale')], limit=1).id
+            return
+
+        addr = self.partner_id.address_get(['delivery', 'invoice', 'default'])
+        pricelist = self.partner_id.property_product_pricelist and self.partner_id.property_product_pricelist.id or False
+
+        self.address_id = addr['delivery'] or addr['default']
+        self.partner_invoice_id = addr['invoice']
+        self.pricelist_id = pricelist
+
+    @api.multi
+    def action_cancel_draft(self):
+        """ Cancels repair order when it is in 'Draft' state.
+        @param *arg: Arguments
+        @return: True
+        """
+        if not self:
+            return False
+        for repair in self:
+            repair.operations_ids.write({'state': 'draft'})
+        self.write({'state': 'draft'})
+        return self.create_workflow()
+
+    @api.multi
+    def action_confirm(self):
+        """ Repair order state is set to 'To be invoiced' when invoice method
+        is 'Before repair' else state becomes 'Confirmed'.
+        @param *arg: Arguments
+        @return: True
+        """
+        for order in self:
+            if (order.invoice_method == 'b4repair'):
+                self.write({'state': '2binvoiced'})
+            else:
+                self.write({'state': 'confirmed'})
+                for line in order.operations_ids:
+                    if line.product_id.tracking != 'none' and not line.lot_id:
+                        raise UserError(_("Serial number is required for operation line with product '%s'") % (line.product_id.name))
+                [line_order.write({'state': 'confirmed'}) for line_order in order.operations_ids]
```

Comment:
???? cannot directly write with order.operations_ids.write({'state': 'confirmed'}) same as you already did above 
action_cancel_draft function.